panic while using named contexts with platforms

buildx v0.8.0-rc1

repro: https://github.com/crazy-max/buildx-buildkit-tests/tree/main/bake-multi

```dockerfile
FROM alpine:3.15
ARG TARGETPLATFORM
WORKDIR /src
RUN echo $TARGETPLATFORM > platform
```
> `baseapp.Dockerfile`

```dockerfile
# syntax=docker/dockerfile-upstream:master-labs
FROM baseapp
RUN cat platform
```
> `Dockerfile`


```dockerfile
# docker-bake.hcl
target "base" {
  dockerfile = "baseapp.Dockerfile"
  args = {
    basefoo = "bar"
  }
  platforms = [
    "linux/amd64",
    "linux/arm64"
  ]
}

target "app" {
  contexts = {
    baseapp = "target:base"
  }
}
```

```console
$ docker buildx bake --metadata-file metadata.json --set app.tags=user/app:latest --set app.output=type=cacheonly app
panic: runtime error: index out of range [0] with length 0

goroutine 157 [running]:
github.com/docker/buildx/build.Build.func4.1()
        /src/build/build.go:762 +0x62b
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /src/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x67
created by golang.org/x/sync/errgroup.(*Group).Go
        /src/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x92
```

--- a/build/build.go
+++ b/build/build.go
@@ -189,6 +189,10 @@ func splitToDriverPairs(availablePlatforms map[string]int, opt map[string]Option
 			pp = append(pp, p)
 			mm[idx] = pp
 		}
+		// if no platform is specified, use first driver
+		if len(mm) == 0 {
+			mm[0] = nil
+		}
 		dps := make([]driverPair, 0, 2)
 		for idx, pp := range mm {
 			dps = append(dps, driverPair{driverIndex: idx, platforms: pp})

